packages/components/src/components/AddressInput/AddressInput.tsx

```diff
@@ -0,0 +1,96 @@
+import { FormControl, FormLabel, Input, FormErrorMessage, InputProps } from '@chakra-ui/react';
+import React, { useEffect, useState } from 'react';
+import { ethers } from 'ethers';
+import { useDebounce } from './useDebounce';
+import { JsonRpcSigner } from '@ethersproject/providers/src.ts/json-rpc-provider';
+
+export interface AddressInputProps {
+  /**
+   * @dev The provider or signer to fetch the address from the ens
+   * @type JsonRpcSigner or ethers.providers.Web3Provider
+   */
+  provider: ethers.providers.Web3Provider | JsonRpcSigner;
+  /**
+   * @dev The value for the input
+   * @type string
+   * @default ''
+   */
+  value: string;
+  /**
+   * @dev Change hanlder for the text input
+   * @type (value: string) => void
+   */
+  onChange: (value: string) => void;
+}
+
+/**
+ * @dev A text input component that is used to get the address of the user from the ens. You can also pass all the styling props of the Chakra UI Input component.
+ * @param provider The provider or signer to fetch the address from the ens
+ * @param value The value for the input
+ * @param onChange Change hanlder for the text input
+ * @returns JSX.Element
+ */
+export const AddressInput: React.FC<AddressInputProps & InputProps> = ({
+  provider,
+  value,
+  onChange,
+  ...props
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const deboucedValued = useDebounce(inputValue, 700);
+  const [, setEnsValue] = useState('');
```

Comment:
Do we need this state setter here?  Since we're not reading the value, do we really need to set it?

Reply:
Yeah, the use of this variable can be avoided, and probably it won't be needed.